src/pipx/interpreter.py

```diff
@@ -97,7 +97,8 @@ def find_python_interpreter(python_version: str, fetch_missing_python: bool = Fa
         if py_executable:
             return py_executable
     except (subprocess.CalledProcessError, FileNotFoundError) as e:
-        raise InterpreterResolutionError(source="py launcher", version=python_version) from e
+        if not fetch_missing_python and not FETCH_MISSING_PYTHON:
+            raise InterpreterResolutionError(source="py launcher", version=python_version) from e
```

Comment:
```suggestion
            raise
```

This should be enough?

Reply:
This will result in a very detailed traceback, which is not user friendly here.

Reply:
Both of them will provide the same output like this:
> $ pipx install black --python 3.7
⚠️ No executable for the provided Python version '3.7' found in py launcher. Please make sure the provided version is
    listed when running `py --list`.